src/main/java/shadows/apotheosis/adventure/affix/socket/SocketHelper.java

```diff
@@ -54,9 +56,18 @@ public static void setGems(ItemStack stack, List<ItemStack> gems) {
 	}
 
 	public static int getSockets(ItemStack stack) {
-		var inst = AffixHelper.getAffixes(stack).get(Affixes.SOCKET.get());
-		if (inst == null) return 0;
-		return (int) inst.level();
+		var socketAffix = AffixHelper.getAffixes(stack).get(Affixes.SOCKET.get());
+		var sockets = 0;
+		if (socketAffix != null) {
+			sockets = (int) socketAffix.level();
+		}
+		var event = new GetItemSocketsEvent(stack, sockets);
+		MinecraftForge.EVENT_BUS.post(event);
+		if (sockets != event.getSockets()) {
+			sockets = event.getSockets();
+			setSockets(stack, sockets);
```

Comment:
NBT Sockets should not be set as the result of the event.
The event should be able to modify the "real" number of sockets used by gameplay, but it should not edit the item NBT - similar to the paradigm used by enchantments and attributes.